src/blockdiagonal.jl

```diff
@@ -152,7 +152,8 @@ function _block_indices(B::BlockDiagonal, i::Integer, j::Integer)
         p += 1
         j -= blocksize(B, p)[2]
     end
-    @views @inbounds i -= sum(size.(blocks(B)[1:(p-1)], 1))
+    # isempty to avoid reducing over an empty collection
+    @views @inbounds i -= isempty(1:(p-1)) ? 0 : sum(size.(blocks(B)[1:(p-1)], 1))
```

Comment:
The alternative is
```suggestion
    @views @inbounds i -= sum(size.(blocks(B), 1)[1:(p-1)])
```
which is cleaner but it allocates more since is builds the array of length `nblocks` rather than just the subset